src/main/java/uk/gov/crowncommercial/dts/scale/cat/service/ProcurementProjectService.java

```diff
@@ -185,4 +187,107 @@ public void updateProcurementProjectName(final Integer projectId, final String p
+    var dbProject = retryableTendersDBDelegate.findProcurementProjectById(projectId)
+        .orElseThrow(() -> new ResourceNotFoundException("Project '" + projectId + "' not found"));
+    var getProjectUri = jaggaerAPIConfig.getGetProject().get(ENDPOINT);
+    var jaggaerProject = ofNullable(jaggaerWebClient.get()
+        .uri(getProjectUri, dbProject.getExternalProjectId()).retrieve().bodyToMono(Project.class)
+        .block(Duration.ofSeconds(jaggaerAPIConfig.getTimeoutDuration())))
+            .orElseThrow(() -> new JaggaerApplicationException(INTERNAL_SERVER_ERROR.value(),
+                "Unexpected error retrieving project"));
+
+
+    // Get Rfx (email recipients)
+    var dbEvent = getCurrentEvent(dbProject);
+    var exportRfxUri = jaggaerAPIConfig.getExportRfx().get(ENDPOINT);
+    var exportRfxResponse =
+        ofNullable(jaggaerWebClient.get().uri(exportRfxUri, dbEvent.getExternalEventId()).retrieve()
+            .bodyToMono(ExportRfxResponse.class)
+            .block(ofSeconds(jaggaerAPIConfig.getTimeoutDuration())))
+                .orElseThrow(() -> new JaggaerApplicationException(INTERNAL_SERVER_ERROR.value(),
+                    "Unexpected error retrieving rfx"));
+
+    // Combine the user IDs and remove duplicates
+    final Set<String> teamIds = jaggaerProject.getProjectTeam().getUser().stream().map(User::getId)
+        .collect(Collectors.toSet());
+    teamIds.addAll(exportRfxResponse.getEmailRecipientList().getEmailRecipient().stream()
+        .map(e -> e.getUser().getId()).collect(Collectors.toSet()));
+
+    // Retrieve additional info on each user from Jaggaer and Conclave
+    return teamIds.stream().map(this::getTeamMember).filter(Objects::nonNull)
+        .collect(Collectors.toSet());
+  }
+
+  /**
+   * Get a Team Member.
+   *
+   * @param jaggaerUserId
+   * @return TeamMember
+   */
+  private TeamMember getTeamMember(final String jaggaerUserId) {
+    try {
+
+      var jaggaerUser = userProfileService.resolveJaggaerUserEmail(jaggaerUserId);
+      var conclaveUser = conclaveService.getUserProfile(jaggaerUser.getEmail());
+
+      var tm = new TeamMember();
+      var cp = new ContactPoint1();
+      // Conclave user id is currently email, this may be changed in the future if the `extCode`
+      // attribute in Jaggaer is used to store a different Conclave id.
+      tm.setId(conclaveUser.getUserName());
+      cp.setName(conclaveUser.getFirstName() + " " + conclaveUser.getLastName());
+      cp.setEmail(conclaveUser.getUserName());
+      cp.setTelephone(jaggaerUser.getPhoneNumber());
+      // TODO: can get more contact info from Conclave via
+      // conclaveService.getUserContacts(jaggaerUser.getEmail())
+      // Question on logic for doing this outstanding on SCAT-2240
+      // cp.setFaxNumber(null);
+      // cp.setUrl(null);
+      tm.setContact(cp);
+      return tm;
+
+    } catch (final WebClientResponseException wcre) {
+      if (wcre.getStatusCode() == HttpStatus.NOT_FOUND) {
```

Comment:
Found the way to do this as part of the webclient call - no need to refactor now, just mentioning it for info, see:
https://github.com/Crown-Commercial-Service/ccs-scale-cat-service/blob/feature/CON-1680_get_users_login_director/src/main/java/uk/gov/crowncommercial/dts/scale/cat/service/ConclaveWrapperApiService.java#L37

Reply:
I've rebased and updated based on this feedback - should be pushed up there